parser/src/main/scala/com/intuit/superglue/pipeline/parsers/CalciteStatementParser.scala

```diff
@@ -12,18 +13,33 @@ import org.apache.calcite.sql.validate.SqlConformanceEnum
 import scala.util.{Failure, Success, Try}
 
 class CalciteStatementParser extends StatementParser {
-  private val calciteConfig: SqlParser.Config = SqlParser.configBuilder()
-    .setParserFactory(SqlDdlParserImpl.FACTORY)
-    .setConformance(SqlConformanceEnum.MYSQL_5)
-    .build()
+  private val defaultCalciteConfig: SqlParser.Config = SqlParser.config()
+    .withParserFactory(SqlDdlParserImpl.FACTORY)
+    .withConformance(SqlConformanceEnum.MYSQL_5)
 
   /**
     * Parses the provided string as SQL using the Calcite parser.
+    * Dialect must be one of [[DatabaseProduct]] if provided. Default to MYSQL_5 mode if not provided.
     *
     * @return The [[StatementMetadataFragment]] created by analyzing the
     *         contents of the SQL statement.
     */
-  override def parseStatement(statement: String): StatementMetadataFragment = {
+  override def parseStatement(statement: String, dialect: Option[String]): StatementMetadataFragment = {
+    // Get DatabaseProduct for the dialect config
+    // For backward compatibility fall back to default (MYSQL_5 SQL compatibility mode and Oracle Lex) if not provided.
+    // Return IllegalArgumentException if an invalid dialect name is provided.
+    val calciteConfig = dialect match {
+      case Some(dialectValue) => Try {
+        DatabaseProduct.valueOf(dialectValue.toUpperCase)
+      } match {
+        case Success(databaseProduct) => databaseProduct.getDialect.configureParser(defaultCalciteConfig)
```

Comment:
So i looked at the calcite source code , and it sets the conformance level based on the config you pass, irrespective of dialect - https://github.com/apache/calcite/blob/master/core/src/main/java/org/apache/calcite/sql/parser/SqlParser.java. They take the dialect into account in unparse method. So that's why it better to set conformance level to BABEL so as to parse queries (supported by any dialect).  
Adding dialect here in config definitely makes sense as we can set different configs per dialect and in future we may also explore query rewrite/unparse capability.